Week2 [STEP 1] Kobe

devKobe24 opened the pull request:
@Gundy93
 건디 잘못된 레포에 PR을 보내 다시 PR을 보냅니다 ㅠㅠ.

 ## 고민했던 점
 ### 누가 봐도 이해 할 수 있도록 순서도를 그리는 것
 - 이해할 수 있게 순서도를 그리고 그 순서도에 알맞은 글을 써 넣도록 고민하고 노력했습니다.
 - 순서도를 그릴때 한 쪽으로 순서가 내려가도록 건디에게 피드백 받은 그대로 할 수 있도록 고민했습니다.

 ---

  ## 해결이 되지 않은 점
  ### 이번 스텝에서는 없었습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 생각만으로 순서도를 그린다는게 쉽지 않았습니다.
 - 건디는 어떤 방법으로 순서도를 그리시나요? 저는 코드를 봐가면서 순서도를 그렸던 적도 있었는데 순서도를 생각으로만 만들려니 매우 어려웠습니다.

Commit:
Week2 [STEP 1] Kobe

 @Gundy93
 건디 잘못된 레포에 PR을 보내 다시 PR을 보냅니다 ㅠㅠ.

 ## 고민했던 점
 ### 누가 봐도 이해 할 수 있도록 순서도를 그리는 것
 - 이해할 수 있게 순서도를 그리고 그 순서도에 알맞은 글을 써 넣도록 고민하고 노력했습니다.
 - 순서도를 그릴때 한 쪽으로 순서가 내려가도록 건디에게 피드백 받은 그대로 할 수 있도록 고민했습니다.

 ---

  ## 해결이 되지 않은 점
  ### 이번 스텝에서는 없었습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 생각만으로 순서도를 그린다는게 쉽지 않았습니다.
 - 건디는 어떤 방법으로 순서도를 그리시나요? 저는 코드를 봐가면서 순서도를 그렸던 적도 있었는데 순서도를 생각으로만 만들려니 매우 어려웠습니다.

Gundy93 commented:
코비님! 바로 다음스텝을 진행하셨군요 멋집니다 👍

> 생각만으로 순서도를 그린다는게 쉽지 않았습니다.

이 부분은 코드가 없이 순서도를 그리는 것이 어려웠다는 말로 이해했는데 맞을까요?

우선 생각이 잘 정리되지 않으면 글로 적어보는 것이 효과적인 것 같습니다.

지난번의 라면 예제와 동일한 이야기인데요

![image](https://user-images.githubusercontent.com/106914201/222671014-dbf438c2-bebf-41ff-b84b-f9525c4729bd.png)

1. 냄비를 준비합니다.
2. 물을 550ml 붓습니다.
3. 불을 킵니다.
4. 기다립니다.
5. 물이 끓는가? - 아니오: 4~5의 과정을 반복합니다.
6. 분말 스프와 후레이크, 면을 넣습니다.
7. 기다립니다.
8. 4분 30초가 되었는가? - 아니오: 7~8의 과정을 반복합니다.
9. 라면을 그릇에 담는다.

(코드적 지식이 배제된) 문장으로 명확한 순서가 정리되었다면 그 다음에 이를 도식화합니다.

![제목 없는 다이어그램 drawio-8](https://user-images.githubusercontent.com/106914201/222673978-b9a1faaa-9b19-426d-b4e9-cea486e49cd1.png)

이러한 흐름으로 만든 순서도를 토대로 또 코드로 옮겨보는 작업이 이 다음 스텝에서 이루어질 것입니다!

현재는 타입 이름도 나오고 변수 이름도 나오고 `for` 문 등등 프로그래밍 경험이 없는 사람은 이해할 수 없는 순서도일 것입니다.

로또를 뽑는 과정을 프로그래밍과 무관한 사람도 설명할 수 있듯이 이 코드 진행도 그러할 것입니다.

가령 친구에게 설명한다고 생각하면 좋지 않을까요?

Commit:
Week2 [STEP 1] Kobe

 @Gundy93
 건디의 피드백대로 수정하니 정말 잘됩니다 :)
 순서도 수정하여 다시 PR 보냅니다

 ## 고민했던 점
 ### 친구에게 설명하듯 순서도를 그리는 것
 - 건디의 피드백처럼 설명서 혹은 친구에게 설명하듯이 순서도를 고민하며 그렸습니다.

 ---

 ## 해결이 되지 않은 점
 ### 이번 스텝에서는 없었습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 순서도와 코딩에 대해서
 - 항상 코딩을 할때 순서도를 그려야 하나요?? 아니면 어느 정도 익숙해지면 머리에서 자동으로 순서도가 그려지나요?? ㅎㅎ
   건디는 코딩을 하기전에 순서도를 그리시는지 궁굼합니다.

Commit:
Week2 [STEP 1] Kobe

 @Gundy93
 건디, 이번에는 친구에게 말하듯 따로 메모장에 적은 후 메모장을 봐가며 순서도를 만들어봤습니다.
 그러니까 이런 순서도가 만들어졌어요!! 순서도 만들기에 재미를 붙여갑니다 ㅎㅎ
 다시 수정본 PR 보냅니다.

 ## 고민했던 점
 ### 코딩용어를 어떻게 쉽게 풀어쓸까
 - 친구나 모르는 사람에게 말하듯 메모장에 먼저 적은 후 순서도를 그렸습니다.

 ---

 ## 해결이 되지 않은 점
 ### 이번 스텝에서는 없었습니다.

 ---

 ## 조언을 얻고 싶은 부분
 ### 조언보다는 소식을 전하려합니다 ㅎㅎ
 - 건디, 오늘 커리어 캠프 신청하려고 합니다.
 벌써부터 긴장이 되네요 ㅠㅠ 제발 테스트에 합격하고 면접도 잘봐서 꼭 붙고싶습니다.
 이렇게 좋은 경험을 스타터 캠프에서 했더니 커리어 캠프에서는 얼마나 더 재미있는 일들이 있을까
 하고는 기대가 됩니다 ㅎㅎ
